go/libraries/doltcore/sqle/dolt_diff_summary_table_function.go

```diff
@@ -0,0 +1,503 @@
+	&sql.Column{Name: "rows_added", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "rows_deleted", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "rows_modified", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "cells_added", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "cells_deleted", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "cells_modified", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "old_row_count", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "new_row_count", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "old_cell_count", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "new_cell_count", Type: sql.Int64, Nullable: true},
+}
+
+// NewInstance implements the TableFunction interface
+func (ds *DiffSummaryTableFunction) NewInstance(ctx *sql.Context, db sql.Database, expressions []sql.Expression) (sql.Node, error) {
+	newInstance := &DiffSummaryTableFunction{
+		ctx:      ctx,
+		database: db,
+	}
+
+	node, err := newInstance.WithExpressions(expressions...)
+	if err != nil {
+		return nil, err
+	}
+
+	return node, nil
+}
+
+// Database implements the sql.Databaser interface
+func (ds *DiffSummaryTableFunction) Database() sql.Database {
+	return ds.database
+}
+
+// WithDatabase implements the sql.Databaser interface
+func (ds *DiffSummaryTableFunction) WithDatabase(database sql.Database) (sql.Node, error) {
+	ds.database = database
+	return ds, nil
+}
+
+// FunctionName implements the sql.TableFunction interface
+func (ds *DiffSummaryTableFunction) FunctionName() string {
+	return "dolt_diff_summary"
+}
+
+// Resolved implements the sql.Resolvable interface
+func (ds *DiffSummaryTableFunction) Resolved() bool {
+	if ds.tableNameExpr != nil {
+		return ds.fromCommitExpr.Resolved() && ds.toCommitExpr.Resolved() && ds.tableNameExpr.Resolved()
+	}
+	return ds.fromCommitExpr.Resolved() && ds.toCommitExpr.Resolved()
+}
+
+// String implements the Stringer interface
+func (ds *DiffSummaryTableFunction) String() string {
+	if ds.tableNameExpr != nil {
+		return fmt.Sprintf("DOLT_DIFF_SUMMARY(%s, %s, %s)", ds.fromCommitExpr.String(), ds.toCommitExpr.String(), ds.tableNameExpr.String())
+	}
+	return fmt.Sprintf("DOLT_DIFF_SUMMARY(%s, %s)", ds.fromCommitExpr.String(), ds.toCommitExpr.String())
+}
+
+// Schema implements the sql.Node interface.
+func (ds *DiffSummaryTableFunction) Schema() sql.Schema {
+	return diffSummaryTableSchema
+}
+
+// Children implements the sql.Node interface.
+func (ds *DiffSummaryTableFunction) Children() []sql.Node {
+	return nil
+}
+
+// WithChildren implements the sql.Node interface.
+func (ds *DiffSummaryTableFunction) WithChildren(children ...sql.Node) (sql.Node, error) {
+	if len(children) != 0 {
+		panic("unexpected children")
+	}
+	return ds, nil
+}
+
+// CheckPrivileges implements the interface sql.Node.
+func (ds *DiffSummaryTableFunction) CheckPrivileges(ctx *sql.Context, opChecker sql.PrivilegedOperationChecker) bool {
```

Comment:
Privs check for this one is another interesting area... I think you made a good choice by requiring select for the affected tables (which is potentially all user tables). 

This would be a really great point to mention in the function docs PR.